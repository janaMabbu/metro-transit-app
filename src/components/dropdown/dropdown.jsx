import React from 'react'
import'./dropdown.less'

export const DropDown = (props) => {
  return (
    <div>
    <select className="metro-app__dropdown" onChange={props.onChange} value={props.selected}>
        <option className="metro-app__dropdown-item">{ props.default }</option>
        { props.context ==='routes' && props.options.map(route => <option className="metro-app__dropdown-item" key={route.get('RouteId')} value={route.get('RouteId')} >{route.get('Description')}</option>) }
        { props.context ==='directions' && props.options.map(direction => <option className="metro-app__dropdown-item" key={direction.get('DirectionId')} value={direction.get('DirectionId')} >{direction.get('DirectionName')}</option>) }
    </select>
    </div>
  )
}