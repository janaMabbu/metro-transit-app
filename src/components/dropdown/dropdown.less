@import "~less/common.less";

.metro-app {
  &__dropdown {
      display: inline-block;
      width: 100%;
      max-width: 420px;
      height: 50px;
      padding: 10px 20px 10px 5px;
      font-size: 20px;
      font-weight: 400;
      line-height: 1.5;
      color: @lightest-grey;
      vertical-align: middle;
      background: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4 5'%3e%3cpath fill='%23343a40' d='M2 0L0 2h4zm0 5L0 3h4z'/%3e%3c/svg%3e") no-repeat right 0.75rem center/8px 10px;
      background-color: @white-human;
      border: 1px solid @white-human2;
      border-radius: 0;
      -webkit-appearance: none;
      -moz-appearance: none;
      appearance: none;
      margin: 10px 0;
  }

}