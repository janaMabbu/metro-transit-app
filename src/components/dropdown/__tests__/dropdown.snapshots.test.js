import { shallow } from 'enzyme'
import React from 'react'
import Immutable from 'immutable'

import { DropDown } from 'components/dropdown/dropdown'

describe('Component - DropDown', () => {
  let props
  const dropDownMockData = [
    {
      'RouteId': '1',
      'Description': 'Description'
    },
     {
      'RouteId': '2',
      'Description': 'Description11'
    },
     {
      'RouteId': '3',
      'Description': 'Description22'
    },
  ]
  beforeEach(() => {
    props = {
      options: Immutable.fromJS(dropDownMockData),
      context:'routes',
      default:'default',
      onChange: jest.fn()
    }
  })
  it('should render the DropDown snapshot', () => {
    const wrapper = shallow(<DropDown {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('it should have correct nummber of options', () => {
    const wrapper = shallow(<DropDown {...props} />)
    expect(wrapper.find('.metro-app__dropdown-item').length).toBe(4)
    dropDownMockData.pop()
    wrapper.setProps({ options: Immutable.fromJS(dropDownMockData) })
    expect(wrapper.find('.metro-app__dropdown-item').length).toBe(3)
  })

})