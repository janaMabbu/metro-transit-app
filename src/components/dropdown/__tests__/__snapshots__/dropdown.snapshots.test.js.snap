// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - DropDown should render the DropDown snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropDown
    context="routes"
    default="default"
    onChange={[MockFunction]}
    options={
      Immutable.List [
        Immutable.Map {
          "RouteId": "1",
          "Description": "Description",
        },
        Immutable.Map {
          "RouteId": "2",
          "Description": "Description11",
        },
        Immutable.Map {
          "RouteId": "3",
          "Description": "Description22",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <select
        className="metro-app__dropdown"
        onChange={[MockFunction]}
      >
        <option
          className="metro-app__dropdown-item"
        >
          default
        </option>
        Immutable.List [
          <option
            className="metro-app__dropdown-item"
            value="1"
          >
            Description
          </option>,
          <option
            className="metro-app__dropdown-item"
            value="2"
          >
            Description11
          </option>,
          <option
            className="metro-app__dropdown-item"
            value="3"
          >
            Description22
          </option>,
        ]
      </select>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            className="metro-app__dropdown-item"
          >
            default
          </option>,
          Immutable.List [
            <option
              className="metro-app__dropdown-item"
              value="1"
            >
              Description
            </option>,
            <option
              className="metro-app__dropdown-item"
              value="2"
            >
              Description11
            </option>,
            <option
              className="metro-app__dropdown-item"
              value="3"
            >
              Description22
            </option>,
          ],
          false,
        ],
        "className": "metro-app__dropdown",
        "onChange": [MockFunction],
        "value": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "default",
            "className": "metro-app__dropdown-item",
          },
          "ref": null,
          "rendered": "default",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Description",
            "className": "metro-app__dropdown-item",
            "value": "1",
          },
          "ref": null,
          "rendered": "Description",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "Description11",
            "className": "metro-app__dropdown-item",
            "value": "2",
          },
          "ref": null,
          "rendered": "Description11",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "host",
          "props": Object {
            "children": "Description22",
            "className": "metro-app__dropdown-item",
            "value": "3",
          },
          "ref": null,
          "rendered": "Description22",
          "type": "option",
        },
        false,
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <select
          className="metro-app__dropdown"
          onChange={[MockFunction]}
        >
          <option
            className="metro-app__dropdown-item"
          >
            default
          </option>
          Immutable.List [
            <option
              className="metro-app__dropdown-item"
              value="1"
            >
              Description
            </option>,
            <option
              className="metro-app__dropdown-item"
              value="2"
            >
              Description11
            </option>,
            <option
              className="metro-app__dropdown-item"
              value="3"
            >
              Description22
            </option>,
          ]
        </select>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              className="metro-app__dropdown-item"
            >
              default
            </option>,
            Immutable.List [
              <option
                className="metro-app__dropdown-item"
                value="1"
              >
                Description
              </option>,
              <option
                className="metro-app__dropdown-item"
                value="2"
              >
                Description11
              </option>,
              <option
                className="metro-app__dropdown-item"
                value="3"
              >
                Description22
              </option>,
            ],
            false,
          ],
          "className": "metro-app__dropdown",
          "onChange": [MockFunction],
          "value": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "default",
              "className": "metro-app__dropdown-item",
            },
            "ref": null,
            "rendered": "default",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Description",
              "className": "metro-app__dropdown-item",
              "value": "1",
            },
            "ref": null,
            "rendered": "Description",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Description11",
              "className": "metro-app__dropdown-item",
              "value": "2",
            },
            "ref": null,
            "rendered": "Description11",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "host",
            "props": Object {
              "children": "Description22",
              "className": "metro-app__dropdown-item",
              "value": "3",
            },
            "ref": null,
            "rendered": "Description22",
            "type": "option",
          },
          false,
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
